### ISSUE TITLE
Bootstrapped status not properly handled

### ISSUE THREAD

**Issue by snazy** [apache/polaris#560]
### Describe the bug

The "bootstrapped" status is incorrectly assumed to be true after a single request, despite asking to run `bootstrap`.

Repro:
1. Add necessary dependencies to `polaris-dropwizard-service`
  ```
  runtimeOnly(project(":polaris-eclipselink"))
  runtimeOnly(libs.h2)
  runtimeOnly(libs.postgresql)
  ```
2. Starting Polaris w/ eclipselink... using `./gradlew :polaris-dropwizard-service:runApp`
3. `curl http://127.0.0.1:8181/api/catalog/v1/config` --> HTTP/500 / "Error 500 java.lang.IllegalStateException: Realm is not bootstrapped, please run server in bootstrap mode." --> OK
4. Another `curl http://127.0.0.1:8181/api/catalog/v1/config` --> HTTP/401 Unauthorized

**Comment by MonkeyCanCode** [comment]
@snazy here is the fix for this one as well as root cause analysis: https://github.com/apache/polaris/pull/579

**Comment by dimas-b** [comment]
I was not able to reproduce it with current master (5761b53423a5e9f23b10c3d3d2bd03602e07890e) when running from the jar.

```
java -jar dropwizard/service/build/libs/polaris-dropwizard-service-1.0.0-incubating-SNAPSHOT.jar server ./polaris-server.yml
```

I get 401 right away (not 500).

**Comment by dimas-b** [comment]
I was able to reproduce with #581

**Comment by adutra** [comment]
Fixed by #579.
